core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -386,7 +386,9 @@ public void load(PluginWrapper wrapper) throws IOException {
             // initialize plugin
             try {
                 Plugin plugin = wrapper.getPlugin();
-                plugin.setServletContext(pluginManager.context);
+                if (plugin != null) {
```

Comment:
Throw exception if null? I doubt we want to proceed with starting the plugin in such conditions